docs/test_plans/0002-Task_Ingestion.md

```diff
@@ -0,0 +1,128 @@
+
+1. Add a new endpoint where the body should contain the number of tasks.
+2. Open an MR to app-interface setting the same CPU request and limit. Do the same to the memory parameter.
+3. After got it merged, check the pulp-perf namespace and see if the new deployment happened [here](https://console-openshift-console.apps.rhperfcluster.ptjz.p1.openshiftapps.com/k8s/ns/pulp-perf/apps~v1~Deployment)
+4. You will need to access a pod with shell permissions to run the test. Check with @pablomh or @jsmejkal on #pulp-perf-experiment
+5. The response from the API should contain the number of tasks that the api process were able to run.
+
+
+## Results
+Date of the test: YYYY/mm/dd
+
+A possible template for the table could be:
+
+| Run            | Tasks           | Observations          |
+|----------------|-----------------|-----------------------|
+| 1              | {{value}}       | `None`                |
+| 2              | {{value}}       | `...`                 |
+| 3              | {{value}}       | `...`                 |
+| 4              | {{value}}       | `...`                 |
+
+**Key:**
+- Use `-` for unavailable metrics.
+
+## An example to calculate the number of tasks ingested
+```python
+import requests
+import time
+from concurrent.futures import ThreadPoolExecutor, TimeoutError
+
+def send_request_and_process_tasks(url):
+    """
+    Sends a request to the specified URL and calculates the number of tasks processed
+    within a 25-second timeout based on the response.
+    
+    Args:
+        url (str): The endpoint to send the request to.
+    
+    Returns:
+        int: Number of tasks processed, or None if the request fails or times out.
+    """
+    try:
+        start_time = time.time()
+        
+        # Send the request (adjust method, headers, or data as needed)
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an error for bad status codes
+        
+        tasks = response.json().get('tasks')
+        
+        elapsed_time = time.time() - start_time
+        
+        print(f"Processed {tasks} tasks in {elapsed_time:.2f} seconds.")
+        return tasks
+    
+    except requests.exceptions.RequestException as e:
+        print(f"Request failed: {e}")
+        return None
+    except Exception as e:
+        print(f"Error processing tasks: {e}")
+        return None
+
+def run_with_timeout(url, timeout=25):
+    """
+    Runs the request and task processing with a timeout.
+    
+    Args:
+        url (str): The endpoint to send the request to.
+    
+    Returns:
+        int: Number of tasks processed, or None if timeout or error occurs.
+    """
+    with ThreadPoolExecutor(max_workers=1) as executor:
+        future = executor.submit(send_request_and_process_tasks, url)
+        try:
+            result = future.result()
+            return result
+        except TimeoutError:
+            print("Operation timed out after 25 seconds.")
+            return None
```

Comment:
I think the client code should wait indefinitely and the time limits are entirely server side.

Reply:
I think maybe I'm confused on what this is measuring, apologies if I am. What made sense to me was:
1. the client waits indefinitely
2. the threadpool executes a lot of "dispath and wait" functions (each in it's own thread) so the parallelization needs to be high on the threadpool
3. The timing element in this client code should be for the entire test duration so I expected tasks to be dispatched with a 25 second value, but the client code to have something like a 300 second (5 minute) value.